File: 7. Reverse Integer.java
public class Solution {
    public int reverse(int x) {
        
        int sign = (x > 0 ? 1 : -1);
        x *= sign;
        int ans = 0;
        
        do {
            if (ans > (Integer.MAX_VALUE - x % 10) / 10)
                return 0;  
            ans = ans * 10 + x % 10;
            x /= 10;
        } while (x > 0);
        
        return ans * sign;
    }
}

File: 7. Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        int rev = 0; 
        int sign = x < 0 ? -1 : 1;
        x *= sign;
        while(x > 0) {
            if(rev > (INT_MAX - x % 10) / 10)
                return 0;
            rev *= 10;
            rev += x % 10;
            x = x / 10;
        }
        return sign * rev;
    }
};